applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1004,8 +977,62 @@ static void discover_cb(struct bt_conn *conn, int err, enum bt_audio_dir dir)
 		if (valid_codec_cap_check(unicast_server->sink_codec_cap,
 					  temp_cap[temp_cap_index].num_caps, BT_AUDIO_DIR_SINK,
 					  idx.lvl3)) {
-			bt_audio_codec_allocation_set(&lc3_preset_sink.codec_cfg,
-						      unicast_server->location);
+
+			ret = bt_audio_codec_cfg_set_val(&lc3_preset_sink.codec_cfg,
+							 BT_AUDIO_CODEC_CFG_CHAN_ALLOC,
+							 (uint8_t *)&unicast_server->location,
+							 sizeof(unicast_server->location));
+			if (ret < 0) {
+				LOG_ERR("Failed to set codec channel allocation: %d", ret);
+				return;
+			}
+
+			const uint8_t *codec_frame_len_ptr = NULL;
+
+			ret = bt_audio_codec_cfg_get_val(&lc3_preset_sink.codec_cfg,
+							 BT_AUDIO_CODEC_CFG_FRAME_LEN,
+							 &codec_frame_len_ptr);
+			if (ret < 0) {
+				LOG_ERR("Failed to get codec frame length: %d", ret);
+				return;
+			}
+
+			uint16_t cap_max_frame_len = 0;
+			uint16_t codec_frame_len = sys_get_le16(codec_frame_len_ptr);
+
+			for (int i = 0; i < temp_cap[temp_cap_index].num_caps; i++) {
+				uint16_t cap_max_frame_len_temp = 0;
+				const uint8_t *cap_max_frame_len_temp_ptr = NULL;
+
+				ret = bt_audio_codec_cap_get_val(&temp_cap[temp_cap_index].codec[i],
+								 BT_AUDIO_CODEC_CFG_FRAME_LEN,
+								 &cap_max_frame_len_temp_ptr);
+				if (ret < 0) {
+					LOG_ERR("Failed to get codec cap frame length: %d", ret);
+					return;
+				}
+
+				/* The first 16 bits are the min frame length */
```

Comment:
Can we do this a tad more dynamically?

Reply:
Don't know how, since the ltv is just storing the four bytes sequentially, and we have to skip the first 2 bytes. We could store both and then not use the first one, but we still need to know that this is two 16 bits values.